src/api/app/controllers/webui/package_controller.rb

```diff
@@ -984,18 +984,40 @@ def meta
   end
 
   def save_meta
-    begin
-      frontend.put_file(params[:meta], :project => @project, :package => @package, :filename => '_meta')
-    rescue ActiveXML::Transport::Error => e
-      message = e.summary
-      flash[:error] = message
-      @meta = params[:meta]
-      render :text => message, :status => 400, :content_type => 'text/plain'
+    meta_xml = Xmlhash.parse(params[:meta])
+    unless meta_xml
+      render(text: "Invalid XML file", content_type: 'text/plain', status: 400)
```

Comment:
I'd prefer to have an ``error`` variable and just render once at the end of the function!
Something like we did in https://github.com/openSUSE/open-build-service/blob/master/src/api/app/controllers/webui/package_controller.rb#L645
I don't really like that we have 5 render & return in one action!